services/web/server/src/simcore_service_webserver/projects/projects_api.py

```diff
@@ -196,6 +201,53 @@ async def update_project_last_change_timestamp(
     await db.update_project_last_change_timestamp(ProjectIDStr(f"{project_uuid}"))
 
 
+async def patch_project(
+    app: web.Application,
+    *,
+    user_id: UserID,
+    project_uuid: ProjectID,
+    project_patch: ProjectPatch,
+):
+    _project_patch_exclude_unset: dict[str, Any] = jsonable_encoder(
+        project_patch, exclude_unset=True, by_alias=False
+    )
+    db: ProjectDBAPI = app[APP_PROJECT_DBAPI]
+
+    # 1. Get project
+    project_db = await db.get_project_db(project_uuid=project_uuid)
+
+    # 2. Check user permissions
+    _user_project_access_rights = await db.get_user_project_access_rights(
```

Comment:
I was wondering why ```get_user_project_access_rights``` is not raising automatically. I find a bit dangerous to not raise because one could easily go on without checking if it is None. and somehow now you check whether it is None, which actually means that one calling patch_project has absolutely no rights to do anything with this project and the fact that the user has write access.
In this case here this is ok, but I find it makes it a bit easy to forget. What if I get the project access right, and since it does not raise I implicitely think the user has read access?